src/ProjectFileTools.NuGetSearch/Feeds/Web/NuGetV2ServiceFeed.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Newtonsoft.Json.Linq;
+using ProjectFileTools.NuGetSearch.Contracts;
+using ProjectFileTools.NuGetSearch.IO;
+using ProjectFileTools.NuGetSearch.Search;
+
+namespace ProjectFileTools.NuGetSearch.Feeds.Web
+{
+    public class NuGetV2ServiceFeedFactory : IPackageFeedFactory
+    {
+        private readonly IWebRequestFactory _webRequestFactory;
+
+        public NuGetV2ServiceFeedFactory(IWebRequestFactory webRequestFactory)
+        {
+            _webRequestFactory = webRequestFactory;
+        }
+
+        public bool TryHandle(string feed, out IPackageFeed instance)
+        {
+            if (Uri.TryCreate(feed, UriKind.Absolute, out Uri location) && feed.EndsWith("nuget", StringComparison.OrdinalIgnoreCase))
+            {
+                instance = new NuGetV2ServiceFeed(feed, _webRequestFactory);
+                return true;
+            }
+
+            instance = null;
+            return false;
+        }
+    }
+
+    public class NuGetV2ServiceFeed : IPackageFeed
+    {
+        private readonly string _feed;
+        private readonly FeedKind _kind;
+        private readonly IWebRequestFactory _webRequestFactory;
+
+        public NuGetV2ServiceFeed(string feed, IWebRequestFactory webRequestFactory)
+        {
+            _webRequestFactory = webRequestFactory;
+            _feed = feed;
+            _kind = feed.IndexOf("nuget.org", StringComparison.OrdinalIgnoreCase) > -1 ? FeedKind.NuGet : FeedKind.MyGet;
+        }
+
+        public string DisplayName => $"{_feed} (NuGet v2)";
+
+        
+        public async Task<IPackageNameSearchResult> GetPackageNamesAsync(string prefix, IPackageQueryConfiguration queryConfiguration, CancellationToken cancellationToken)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return PackageNameSearchResult.Failure;
+            }
+
+            IReadOnlyList<string> results = new List<string>();
+            string frameworkQuery = !string.IsNullOrEmpty(queryConfiguration.CompatibiltyTarget) ? $"&targetFramework={queryConfiguration.CompatibiltyTarget}" : "";
+            var serviceEndpoint = $"{_feed}/Search()";
+            Func<string, string> queryFunc = x => $"{x}?searchTerm='{prefix}'{frameworkQuery}&includePrerelease={queryConfiguration.IncludePreRelease}";
```

Comment:
Is this intended to exclude semver 2 packages? If not, consider something like:

```suggestion
            Func<string, string> queryFunc = x => $"{x}?searchTerm='{prefix}'{frameworkQuery}&includePrerelease={queryConfiguration.IncludePreRelease}&semVerLevel=2.0.0";
```

For example, the [`NuGet.Packaging`](https://www.nuget.org/packages/NuGet.Packaging/) package has a semver2 version `5.2.0+f6f9fb481b9fbf14154d484dbf376446c9b0d274`. The V2 search API will only return this version if explicitly include semver2 results:

* Include semver2: https://www.nuget.org/api/v2/Search()?searchTerm=nuget.packaging&semVerLevel=2.0.0
* Exclude semver2: https://www.nuget.org/api/v2/Search()?searchTerm=nuget.packaging

Reply:
Makes sense! Added query string.
Thanks for the feedback!